examples/bri-3/src/bri/identity/bpiSubjects/api/subjects.controller.spec.ts

```diff
@@ -32,6 +39,84 @@ describe('SubjectController', () => {
     await app.init();
   });
 
+  describe('getBpiSubjectById', () => {
+    it('should throw NotFound if non existent id passed', () => {
+      // Arrange
+      const nonExistentId = '123';
+
+      // Act and assert
+      expect(async () => {
+        await sController.getBpiSubjectById(nonExistentId);
+      }).rejects.toThrow(new NotFoundException(`Bpi Subject with id: ${nonExistentId} does not exist.`));
+    });
+
+    it('should return the correct bpi subject if proper id passed ', async () => {
+      // Arrange
+      const requestDto = { name: "name", desc: "desc", publicKey: "publicKey"} as CreateBpiSubjectDto;
+
+      const newBpiSubjectId = await sController.createBpiSubject(requestDto);
+
+      // Act
+      const createdBpiSubject = await sController.getBpiSubjectById(newBpiSubjectId);
+
+      // Assert
+      expect(createdBpiSubject.id).toEqual(newBpiSubjectId);
+      expect(createdBpiSubject.name).toEqual(requestDto.name);
+      expect(createdBpiSubject.desc).toEqual(requestDto.desc);
+      expect(createdBpiSubject.publicKey).toEqual(requestDto.publicKey);
+    });
+  });
+
+  describe('getAllBpiSubjects', () => {
+    it('should return emtpy array if not bpi subjects ', async () => {
+      // Act
+      const bpiSubjects = await sController.getAllBpiSubjects();
+
+      // Assert
+      expect(bpiSubjects.length).toEqual(0);
+    });
+
+    it('should return 1 bpi subject if 1 exists ', async () => {
+      // Arrange
+      const requestDto = { name: "name", desc: "desc", publicKey: "publicKey"} as CreateBpiSubjectDto;
```

Comment:
can we mock db and return values for specific tests or we have to create objects like this?
also i think that test that returns 1 subject is redundant if we also have a test that returns 2 subjects below

Reply:
Prisma does not support DB mocking to my knowledge. We could try to get the instance of the mock repo in the test and then fill it directly during Arrange. Will try that as part of some upcoming PR and will refactor if ok.

Reply:
Removed obsolete test